arctic/serialization/numpy_arrays.py

```diff
@@ -203,13 +203,17 @@ def deserialize(self, data, columns=None):
         columns: None, or list of strings
             optionally you can deserialize a subset of the data in the SON. Index
             columns are ALWAYS deserialized, and should not be specified
+        inplace: Convert and remove items from data in-place
+            this will modify data
 
         Returns
         -------
         pandas dataframe or series
         """
         if not data:
             return pd.DataFrame()
+        if not inplace:
+            data = data[:]
```

Comment:
there are some errors in the tests, so I'm thinking this will need to be tweaked a bit more